Blip.HttpClient.Tests/EventTrackServiceUnitTests.cs

```diff
@@ -0,0 +1,183 @@
+﻿using Blip.HttpClient.Exceptions;
+using Blip.HttpClient.Factories;
+using Blip.HttpClient.Services.EventTracker;
+using Lime.Messaging.Resources;
+using Lime.Protocol;
+using NSubstitute;
+using NSubstitute.ExceptionExtensions;
+using Serilog;
+using Shouldly;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Take.Blip.Client;
+using Xunit;
+
+namespace Blip.HttpClient.Tests
+{
+    public class EventTrackServiceUnitTests
+    {
+        private const string RecipientIdentity = "unittests.testingbots@0mn.io";
+        private const string Category = "categoryTest";
+        private const string Action = "actionTest";
+        private readonly Contact contact;
+        private readonly IEventTrackService _eventTrackService;
+        private readonly ILogger _logger;
+
+        public EventTrackServiceUnitTests()
+        {
+            contact = new Contact();
+            var clientFactory = new BlipHttpClientFactory();
+            var sender = clientFactory.BuildBlipHttpClient("dGVzdGluZ2JvdHM6OU8zZEpWbHVaSWZNYmVnOWZaZzM=");
+            _eventTrackService = new EventTrackService(sender);
+            _logger = Substitute.For<ILogger>();
+        }
+
+        #region Methods unit tests
+
+        [Theory]
+        [InlineData("", false)]
+        [InlineData("", true)]
+        [InlineData("dGVzdGluZ2JvdHM6OU8zZEpWbHVaSWZNYmVnOWZaZzM=", false)]
+        public async Task Add_UnitTest(string authKey, bool fireAndForget)
+        {
+            Command addResponse;
+            Identity identity = Identity.Parse(RecipientIdentity);
+
+            if (authKey == "")
+            {
+                addResponse = await _eventTrackService.AddAsync(Category, Action, _logger, identity: identity);
+            }
+            else if (fireAndForget)
+            {
+                addResponse = await _eventTrackService.AddAsync(Category, Action, _logger, contact: contact, fireAndForget: fireAndForget);
```

Comment:
According to the Code Coverage results, this line is not testing the `fireAndForget = true` conditional. Debugging is recommended

Reply:
Solved on new commit